web/modules/custom/ct_github/src/GithubQuery.php

```diff
@@ -130,8 +135,38 @@ public function isUserValid(string $username): bool {
    * API request to get user contributions.
    */
   public function getUserContributions(string $username) {
+    if ($this->client === NULL) {
+      return NULL;
+    }
     $query = $this->getQuery($username);
-    return $this->client->api('graphql')->execute($query);
-  }
+    $maxRetries = 5;
+    $retryCount = 0;
+    $waitTime = 1; // Initial wait time in seconds for backoff
 
+    while ($retryCount < $maxRetries) {
+      try {
+        return $this->client->api('graphql')->execute($query);
+      } catch (Github\Exception\RuntimeException $e) {
+        // Check if the error is due to bad credentials
+        if ($e->getMessage() === 'Bad credentials') {
+          $this->logger->error('GitHub API error: Bad credentials. Please check the credentials.');
+          return NULL;
+        }
```

Comment:
Is it likely that we will get bad credentials here? Shouldn't this have been caught in `authenticate` method itself?